packages/dredd/test/integration/regressions/regression-507-test.js

```diff
@@ -0,0 +1,86 @@
+import { assert } from 'chai';
+
+import Dredd from '../../../lib/Dredd';
+import { runDreddWithServer, createServer } from '../helpers';
+
+const bootstrap = (configureApp) => {
+    return new Promise((resolve, reject) => {
+      const app = createServer();
+      configureApp(app);
+
+      const dredd = new Dredd({
+        options: { path: './test/fixtures/regression-507.yaml' },
+      });
+
+      runDreddWithServer(dredd, app, (...args) => {
+        const [err, runtimeInfo] = Array.from(args);
+
+        if (err) {
+          reject(err);
+        }
+
+        resolve(runtimeInfo);
+      });
+    });
+};
+
+describe('Regression: Issue #507', () => {
+  describe('given actual body matches an enum', () => {
+    let runtimeInfo;
+
+    before(async () => {
+      runtimeInfo = await bootstrap((app) => {
```

Comment:
What if we restructure the API description document to use 1 request and 1 response? I've learned that Dredd would hit both req/res in that case, and we could use the same server instance.